silo-vaults/contracts/MetaMorpho.sol

```diff
@@ -863,11 +890,25 @@ contract MetaMorpho is ERC4626, ERC20Permit, Ownable2Step, Multicall, IMetaMorph
     }
 
     function _update(address from, address to, uint256 value) internal virtual override {
+        // on deposit, claim must be first action, new user should not get reward
+
+        // on withdraw, claim must can be first action, user that is leaving should get rewards
+        // immediate deposit-withdraw operation will not abused it, because before deposit all rewards will be
+        // claimed, so on withdraw on the same block no additional rewards will be generated.
+
+        // transfer shares is basically withdraw->deposit, so claiming rewards should be done before any state changes
+
+        claimRewards();
```

Comment:
Why we claim here? Is it for transfer? It looks like we will claim twice on deposit and withdraw.

Reply:
I moved all claims here. I removed fomr other places, most likely you reviewd code just before my commit.